src/Sarif/Writers/PrereleaseCompatibilityTransformer.cs

```diff
@@ -149,6 +152,67 @@ private static void RemoveToolLanguage(JObject run)
             }
         }
 
+        private static void ModifyReportingDescriptorNameType(JObject run)
+        {
+            // Access and modify run.tool
+            JObject tool = (JObject)run["tool"];
+            ModifyReportingDescriptorNameTypeInTool(tool);
+
+            // Access and modify run.conversion.tool
+            if (run["conversion"] is JObject conversion)
+            {
+                tool = (JObject)conversion["tool"];
+                ModifyReportingDescriptorNameTypeInTool(tool);
+            }
+        }
+
+        private static void ModifyReportingDescriptorNameTypeInTool(JObject tool)
+        {
+            // Access and modify tool.driver
+            if (tool["driver"] is JObject driver)
+            {
+                ModifyReportingDescriptorNameTypeInToolComponent(driver);
+            }
+
+            // Access and modify each item in tool.extensions
+            if (tool["extensions"] is JArray extensions)
+            {
+                foreach (JObject toolComponent in extensions)
+                {
+                    ModifyReportingDescriptorNameTypeInToolComponent(toolComponent);
+                }
+            }
+        }
+
+        private static void ModifyReportingDescriptorNameTypeInToolComponent(JObject toolComponent)
+        {
+            // Access and modify toolComponent.notificationDescriptors
+            if (toolComponent["notificationDescriptors"] is JArray notificationDescriptors)
+            {
+                foreach (JObject reportingDescriptor in notificationDescriptors)
+                {
+                    ModifyReportingDescriptorNameInReportingDescriptor(reportingDescriptor);
+                }
+            }
+
```

Comment:
as discussed, push this structure into your helper to simplify this caller and expand your callee

Reply:
i kept the if condition in the caller

---
In reply to: [260416838](https://github.com/Microsoft/sarif-sdk/pull/1309#discussion_r260416838) [](ancestors = 260416838)